src/components/modal/Modal.tsx

```diff
@@ -8,14 +8,23 @@ type BaseModalProps = {
   forwardRef?: LegacyRef<HTMLDivElement> | null;
   outsideClick: () => void;
   className?: string;
+  disableOutsideClick?: boolean;
 };
 
 export type ModalProps = BaseModalProps & PropsWithChildren;
 
-export function Modal({ open, children, outsideClick, className }: ModalProps) {
+export function Modal({
+  open,
+  children,
+  outsideClick,
+  className,
+  disableOutsideClick
+}: ModalProps) {
```

Comment:
Making outsideClick optional is a better solution than adding another prop I think. Then inside the effect you check "if outsideClick"